### ISSUE TITLE
Rendering of in-line stage directions

### ISSUE THREAD

**Issue by JackWilton1594** [projectLEMDO/lemdoIssues#269]
Hi team,

In-line stage directions are rendering in odd ways. Here, in 1HW, for example:

![Screen Shot 2024-09-03 at 13 25 29](https://github.com/user-attachments/assets/19a2aee6-6af2-4000-9c34-14dc3e27be66)

The stage directions should be rendered in brackets and on the same line as the dialogue. The code is:
`<sp who="#emd1HW_M_Hippolito" xml:id="emd1HW_M_a1_s1_sp6">
                  <speaker>Hippolito</speaker>
                  <l part="I"><stage type="delivery" xml:id="emd1HW_M_a1_s1_sd4">To the Duke</stage> I do arrest thee, murderer.</l>
                  <l part="F"><stage type="delivery" xml:id="emd1HW_M_a1_s1_sd5">To Attendants</stage> Set down,</l>
                  <l><anchor xml:id="emd1HW_M_anc_4332"/>Villains, set down <anchor xml:id="emd1HW_M_anc_37"/>that sorrow<anchor xml:id="emd1HW_M_anc_38"/>; ’tis all mine<anchor xml:id="emd1HW_M_anc_4333"/>.</l>
               </sp>`

Two further examples:

![Screen Shot 2024-09-03 at 13 25 42](https://github.com/user-attachments/assets/6eb0a263-8f22-4bc0-a270-d0db0478baed)
![Screen Shot 2024-09-03 at 13 25 48](https://github.com/user-attachments/assets/23b8d297-071e-4ce6-98d4-ab11311bbb0b)

**Comment by martindholmes** [comment]
The style file modernized.scss line 35 specifies that stage directions are by default `display: block`; stage directions which are direct children of the speech (`<sp>`) element are to be rendered inline. @LEMDO-PM @pszpak: is this the intended behaviour? Or was the intention that `<stage>` elements that are _descendants_ of `<sp>` elements should be inline?

**Comment by JackWilton1594** [comment]
Stage directions which are direct children of `<l>` or `<p>` within `<sp>` should be rendered inline. These are typically delivery (_To X_, _Aside_) and other stage business (_Shakes_, _Rattles_, and _Rolls_).

Stage directions outside of these – but still within `<sp>` – should have their own line. These include entrances, exits, and more complex stage business. These should be flush left, except for exits which are flush right. Example:

`<sp>
<speaker>Somebody</speaker>
<l>Blah blah blah.</l>
<stage type="entrance">Somebody Else enters.</stage>
<l>Oh look, Somebody Else is here!</l>
</sp>`

**Comment by martindholmes** [comment]
That looks like an omission in the modernized.scss file, then. I'll tweak it.

**Comment by LEMDO-PM** [comment]
I'll be on this tomorrow afternoon.

**Comment by LEMDO-PM** [comment]
I've committed a fix for this, and Martin has committed a fix for an XSLT issue to restore parentheses. Leaving this open for now, but will close once we see the new build.

**Comment by LEMDO-PM** [comment]
<img width="799" alt="image" src="https://github.com/user-attachments/assets/ecf304f3-f6c0-4208-8b98-ce7d9a9ea79d">

**Comment by LEMDO-PM** [comment]
This looks good in this last build. Closing this issue :)

**Comment by JanelleJenstad** [comment]
This issue was closed prematurely. We still have the problem that `<stage>` elements inside `<sp>` but outside `<l>` or `<p>` are rendering in ways we don't want. 

Example from emd1HW_M:


```
<l>... without form.</l>
<stage type="entrance">Enter George with nightcap and staff.</stage>
<stage type="business">Candido puts on the nightcap and takes the staff.</stage>
<l>So, so, kind George.... </l>
```

This encoding renders like this:

![Image](https://github.com/user-attachments/assets/f1dc526d-2ea3-4f26-bd8c-cc45d070e1bc)

Please add padding around these stage directions and put each one on its own line. 

To recap: `<stage>` _inside_ `<p>` or `<l>` should be rendered inline. `<stage>` outside `<p>` or `<l>` but inside `<sp>` should be treated as a block-level element.

**Comment by LEMDO-PM** [comment]
@JanelleJenstad Is this what you would like?

![Image](https://github.com/user-attachments/assets/8e4e73e0-8969-4298-9e6a-5062df121c8b)

**Comment by JanelleJenstad** [comment]
Yes, exactly. (@JackWilton1594)

**Comment by LEMDO-PM** [comment]
Great, closing now
